src/main/java/it/gov/pagopa/paymentupdater/controller/PaymentController.java

```diff
@@ -34,10 +34,13 @@ public class PaymentController {
 	@Autowired
 	PaymentService paymentService;
 
+
 	@GetMapping(value = "/check/messages/{messageId}")
 	public ResponseEntity<ApiPaymentMessage> getMessagePayment(@PathVariable String messageId) {
 		return paymentService.findById(messageId)
 				.map(pay -> ApiPaymentMessage.builder().messageId(pay.getId())
+						.dueDate(pay.getDueDate() == null ? null : LocalDate.ofInstant(Instant.ofEpochMilli(pay.getDueDate().longValue()), 
+				                TimeZone.getDefault().toZoneId()))
```

Comment:
why not using the second one with Optional directly?

Reply:
fixed, we currently save the dueDate as LocalDateTime, no longer as a long. This helps data operations on the scheduler side.